packages/graphiql-build-fetcher/src/lib.ts

```diff
@@ -0,0 +1,144 @@
+import { DocumentNode, visit } from 'graphql';
+import fetchMultipart from 'fetch-multipart-graphql';
+import { createClient, Client } from 'graphql-ws';
+import { createClient as createLegacyClient } from 'graphql-transport-ws';
+import { makeAsyncIterableIteratorFromSink } from '@n1ru4l/push-pull-async-iterable-iterator';
+
+import type { BuildFetcherOptions } from './';
+import type {
+  Fetcher,
+  FetcherResult,
+  FetcherParams,
+  FetcherOpts,
+} from '@graphiql/toolkit';
+
+/**
+ * Returns true if the name matches a subscription in the AST
+ *
+ * @param document {DocumentNode}
+ * @param name the operation name to lookup
+ * @returns {boolean}
+ */
+export const isSubcriptionWithName = (
+  document: DocumentNode,
+  name: string,
+): boolean => {
+  let isSubcription = false;
+  visit(document, {
+    OperationDefinition(node) {
+      if (name === node.name?.value) {
+        if (node.operation === 'subscription') {
+          isSubcription = true;
+        }
+      }
+    },
+  });
+  return isSubcription;
+};
+
+/**
+ * create a websockets client, following
+ * `graphql-ws` Client signature
+ *
+ * @param options
+ * @returns {Client | null}
+ */
+export const createWebsocketsClient = (
+  options: BuildFetcherOptions,
+): Client | null => {
+  let wsClient: Client | null = null;
+  try {
+    try {
+      // TODO: defaults?
+      wsClient = createClient({
```

Comment:
@enisdenjo so, this is the default pattern, if only `subscriptionsUrl` is provided. the user can optionally provide their own instance of a client, which would bypass this behavior entirely. 

the idea of a `wsClientOptions` option came to mind, however a `wsClient` option itself seemed even more powerful. do you think a `wsClientOptions` with `graphql-ws`'s `ClientOptions` should be offered as well, here?

Reply:
Other runtime options can be configured and I would recommend allowing the user to do so. You may have a custom server configuration and might need to fine tune the client so that they work in coherence.

But, this can be achieved by providing your own client instance, so, `subscriptionsUrl` + `wsClient` would be just enough.

Reply:
Thinking about it a bit more, I'd expose the `connectionParams` too. I see it being used a lot because it is the "headers" of the WS connection.

Reply:
ah, fantastic idea!